docs/application/dotnet/guides/user-interface/nui/vectorgraphics/ShapeAndStroke.md

```diff
@@ -0,0 +1,255 @@
+The example below shows how to create a circle with a shading effect.
+
+```cs
+//generate a shape
+Shape circle = new Shape();
+
+//add a circle to the shape (cx, cy, rx, ry)
+circle.AddCircle(300.0f, 300.0f, 100.0f, 100.0f);
+
+//generate a radial gradient
+RadialGradient radialGradient = new RadialGradient();
+
+//set the radial gradient geometry info (center point, radius)
+radialGradient.SetBounds(new Position2D(300, 300), 150.0f);
+
+//gradient colors
+List<ColorStop> colorStops = new List<ColorStop>() {
+                                new ColorStop(0.0f, new Color(1.0f, 1.0f, 1.0f, 1.0f)),
+                                new ColorStop(1.0f, new Color(1.0f, 0.5f, 0.5f, 1.0f)) };
+
+radialGradient.ColorStops = colorStops.AsReadOnly();
+
+//set the shape radial gradient
+circle.FillGradient = radialGradient;
+
+//add the shape into the canvas view
+canvasView.AddDrawable(circle);
+```
+
+![ShapeAandStroke_Gradient](./media/vectorgraphics_gradient.png)
+
+By setting the `Spread` with `SpreadType`, you can create various gradients style.
+
+| SpreadType | Description                                                                                                 |
+| -----------| ----------------------------------------------------------------------------------------------------------- |
+| `Pad`      | The remaining area is filled with the closest stop color.                                                   |
+| `Reflect`  | The gradient pattern is reflected outside the gradient area until the expected region is filled.            |
+| `Repeat`   | The gradient pattern is repeated continuously beyond the gradient area until the expected region is filled. |
+
+```cs
+//generate a shape
+Shape rect = new Shape();
+
+//add a rectangle to the shape (cx, cy, rx, ry)
+rect.AddRect(50, 50, 200, 200, 20, 20);
+
+//generate a linear gradient
+LinearGradient linearGradient = new LinearGradient();
+
+//set the linear gradient geometry info (first point, second point)
+linearGradient.SetBounds(new Position2D(100, 100), new Position2D(200, 200));
+
+//gradient colors
+List<ColorStop> colorStops = new List<ColorStop>() {
+                                new ColorStop(0.0f, new Color(1.0f, 1.0f, 1.0f, 1.0f)),
+                                new ColorStop(1.0f, new Color(0.0f, 0.5f, 1.5f, 1.0f)) };
+
+linearGradient.ColorStops = colorStops.AsReadOnly();
+
+//set the linear gradient spread type.
+linearGradient.Spread = SpreadType.Reflect;
+
+//set the shape linear gradient
+rect.FillGradient = linearGradient;
+
+//add the shape into the canvas view
+canvasView.AddDrawable(rect);
+```
+The output of the example is as follows: <br>
+![ShapeAandStroke_Gradient2](./media/vectorgraphics_gradient2.png)
+<br>
+
+
+## Stroking
+Stroking enables you to draw the outline of shapes as well lines. You can simply add stroke properties to a shape if needed.<br>
```

Comment:
The meaning is not clear here... "Stroking enables you to draw the outline of shapes as well lines". Should it be "...of shapes as well as lines"?
```suggestion
Stroking enables you to draw the outline of shapes as well lines. You can simply add stroke properties to a shape if needed.<br>
```